### PR TITLE
Added constructors with name argument to the target types

### PR DESCRIPTION
**Pull request by flyingcroissant**
Resolution for #1297. Seems to build with Appveyor. Let me know if there are any issues.

### CODE REVIEW

**Comment by 304NotModified** [comment]
\0/

**Comment by 304NotModified** [comment]
Thanks also for the unit test, but locally I get a stackoverflow? 

image

**Comment by 304NotModified** [comment]
update that is fixed after rebase

**Comment by 304NotModified** [comment]
this need some change as the current unit test wont fully work. Trying to fix it :)

`Typeof(Target).GetProperties()` was only returning "Name", which was excluded in the test case ;)

**Comment by flyingcroissant** [comment]
I guess I should have checked if there were any other public properties, haha :)  The Type.GetProperties MSDN documentation says you can use BindingFlags parameters to change what it returns (eg. BindingFlags.NonPublic | BindingFlags.Instance | BindingFlags.Static). That set of flags seems to give the protected properties as well.

When I make that change, the test fails because (at least) the AspResponseTarget.IsInitialized property is different with the two constructors. I haven't looked into it too much yet (it looks like the same thing should be happening for both?), but should I be calling Initialize() on each Target before comparing?

**Comment by flyingcroissant** [comment]
Also FYI, I had to filter out the SyncRoot property in the test since that's expected to be a different object in both. Maybe there's a smarter way to do that filtering and handle future property additions.

**Comment by flyingcroissant** [comment]
I think I fixed the other issue I was having too. It was because the assertion was comparing the property value from one instance against the other instance (ie. not the property value from that instance).

File: tests/NLog.UnitTests/Targets/TargetTests.cs [hunk 1]
```diff
@@ -65,13 +65,16 @@ public void ConstructorsTest()
                         namedConstructedTarget = (Target)Activator.CreateInstance(candidateType, candidateType.ToString());
 
                         // Check that the created targets are the same except for name
-                        foreach (System.Reflection.PropertyInfo pi in (typeof(Target)).GetProperties())
+                        foreach (System.Reflection.PropertyInfo pi in (typeof(Target)).GetProperties(
```
Comment (304NotModified) [inline tests/NLog.UnitTests/Targets/TargetTests.cs]: `typeof(Target)` => `typeof(candidateType)` ?

**Comment by 304NotModified** [comment]
I see you have changed the test, but the test aren't working now, as it should fail (there are some `this()` calls missing in the PR)

**Comment by 304NotModified** [comment]
@flyingcroissant see https://github.com/flyingcroissant/NLog/pull/1

**Comment by flyingcroissant** [comment]
Thanks, I've merged in your changes. I'll see what I can think of for doing the property comparisons.

**Comment by 304NotModified** [comment]
superseded by https://github.com/NLog/NLog/pull/1434

**Pull request closed without merging**